mimic/test/test_maas.py

```diff
@@ -400,6 +431,272 @@ def test_delete_alarm(self):
+        """
+        req = request(self, self.root, "PUT",
+                      '{0}/entities/{1}/checks/test_responses/{2}'.format(
+                          self.ctl_uri, self.entity_id, 'agent.memory'),
+                      json.dumps([{'status': 'whuuut'}]))
+        resp = self.successResultOf(req)
+        self.assertEquals(resp.code, 204)
+        req = request(self, self.root, "POST",
+                      self.uri + '/entities/' + self.entity_id + '/test-check',
+                      json.dumps({'type': 'agent.memory'}))
+        resp = self.successResultOf(req)
+        self.assertEquals(resp.code, 200)
+        data = self.get_responsebody(resp)
+        self.assertEquals('whuuut', data[0]['status'])
+
+    def test_test_check_setting_metrics(self):
+        """
+        The test-check control API can set metrics.
+
+        Subsequent requests to set the same metrics on the same check type
+        for the same entity will override.
+        """
+        req = request(self, self.root, "PUT",
+                      '{0}/entities/{1}/checks/test_responses/{2}'.format(
+                          self.ctl_uri, self.entity_id, 'remote.http'),
+                      json.dumps([{'metrics': {'duration': {'data': 123}},
+                                   'monitoring_zone_id': 'mzdfw'}]))
+        resp = self.successResultOf(req)
+        self.assertEquals(resp.code, 204)
+        req = request(self, self.root, "POST",
+                      self.uri + '/entities/' + self.entity_id + '/test-check',
+                      json.dumps({'type': 'remote.http',
+                                  'monitoring_zones_poll': ['mzdfw']}))
+        resp = self.successResultOf(req)
+        self.assertEquals(resp.code, 200)
+        data = self.get_responsebody(resp)
+        self.assertEquals(123, data[0]['metrics']['duration']['data'])
+        req = request(self, self.root, "PUT",
+                      '{0}/entities/{1}/checks/test_responses/{2}'.format(
+                          self.ctl_uri, self.entity_id, 'remote.http'),
+                      json.dumps([{'metrics': {'duration': {'data': 456}},
+                                   'monitoring_zone_id': 'mzdfw'}]))
+        resp = self.successResultOf(req)
+        self.assertEquals(resp.code, 204)
+        req = request(self, self.root, "POST",
+                      self.uri + '/entities/' + self.entity_id + '/test-check',
+                      json.dumps({'type': 'remote.http',
+                                  'monitoring_zones_poll': ['mzdfw']}))
+        resp = self.successResultOf(req)
+        self.assertEquals(resp.code, 200)
+        data = self.get_responsebody(resp)
+        self.assertEquals(456, data[0]['metrics']['duration']['data'])
+
+    def test_test_check_clears_metrics(self):
+        """
+        The test-check control API can clear metrics.
+
+        NB: Randomly generated string metrics are between 12 and 30
```

Comment:
The ReST syntax for this would be [`..note:`](http://docutils.sourceforge.net/docs/ref/rst/directives.html#admonitions).